src/main/kotlin/com/ampnet/reportservice/service/data/TransactionFactory.kt

```diff
@@ -15,15 +15,15 @@ const val LENGTH_OF_PERCENTAGE = 8
 
 class TransactionFactory private constructor() {
     companion object {
-        fun createTransaction(transaction: TransactionInfo): Transaction? {
+        fun createTransaction(transaction: TransactionInfo, translations: Translations): Transaction? {
```

Comment:
Translations should not be in the constructor. I would rather move to function or combine with function `setLanugage`